subprojects/execution/src/test/groovy/org/gradle/internal/execution/steps/CreateOutputsStepTest.groovy

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.steps
+
+import org.gradle.api.internal.file.collections.ImmutableFileCollection
+import org.gradle.internal.execution.Context
+import org.gradle.internal.execution.Result
+import org.gradle.internal.execution.UnitOfWork
+import org.gradle.internal.file.TreeType
+
+class CreateOutputsStepTest extends StepSpec {
```

Comment:
So, this is a unit test for `CreateOutputsStep`. I would still like to have a test for checking if the output is actually created when we use the full execution chain. This test would test the integration of the tests between each other, and does not test the many ramifications (like different types of outputs), but only if everything works well together.

In my mind, that is the role of the current `IncrementalExecutionTest`. Do you agree?

Reply:
More or less, yes. We also have `ExecuteActionsTaskExecuterTest`.

BTW, I just moved the existing test from `IncrementalExecutionTest` here. We should definitely add more tests here, for example for incorrect pre-existing output types.

Reply:
I re-added the test for creating outputs to `IncrementalExecutionTest`.